goosebit/ui/bff/software/routes.py

```diff
@@ -64,20 +64,20 @@ async def post_update(
         await create_software_update(url, None)
     else:
         # local file
-        file = config.artifacts_dir.joinpath(filename)
-        config.artifacts_dir.mkdir(parents=True, exist_ok=True)
+        artifacts_dir = Path(config.artifacts_dir)
+        file = artifacts_dir.joinpath(filename)
+        await artifacts_dir.mkdir(parents=True, exist_ok=True)
 
         temp_file = file.with_suffix(".tmp")
         if init:
-            temp_file.unlink(missing_ok=True)
+            await temp_file.unlink(missing_ok=True)
 
-        contents = await chunk.read()
-
-        async with aiofiles.open(temp_file, mode="ab") as f:
-            await f.write(contents)
+        async with await open_file(temp_file, "ab") as f:
```

Comment:
I think we should confirm a preference between aiofiles and anyio for file handling across the project.  I'm alright with using anyio for file handling, but in that case we should work towards complete removal of aiofiles.

Reply:
We can look at this in later versions, without tempfile support this is likely not possible yet.